pkg/nmstate/shared.go

```diff
@@ -0,0 +1,25 @@
+package nmstate
+
+// DesiredState provides struct for the NMState desired state object containing all NMState configuration.
+type DesiredState struct {
+	Interfaces []Interface `yaml:"interfaces"`
+}
+
+// Interface provides struct for the NMState interface state object containing interface information.
+type Interface struct {
```

Comment:
Please rename struct to NetworkInterface in order to avoid name overlapping with golang interface

Reply:
renamed